rocketpy/sensitivity/sensivity_model.py

```diff
@@ -0,0 +1,515 @@
+
+        return
+
+    def set_parameters_nominal(
+        self,
+        parameters_nominal_mean,
+        parameters_nominal_sd,
+    ):
+        """Set parameters nominal mean and standard deviation
+
+        Parameters
+        ----------
+        parameters_nominal_mean : np.array
+            An array contaning the nominal mean for parameters in the
+            order specified in parameters names at initialization
+        parameters_nominal_sd : np.array
+            An array contaning the nominal standard deviation for
+            parameters in the order specified in parameters names at
+            initialization
+        """
+        if len(parameters_nominal_mean) != self.n_parameters:
+            raise ValueError(
+                "Nominal mean array length does not match number of parameters passed at initilization."
+            )
+        if len(parameters_nominal_sd) != self.n_parameters:
+            raise ValueError(
+                "Nominal sd array length does not match number of parameters passed at initilization."
+            )
+
+        for i in range(self.n_parameters):
+            parameter = self.parameters_names[i]
+            self.parameters_info[parameter]["nominal_mean"] = parameters_nominal_mean[i]
+            self.parameters_info[parameter]["nominal_sd"] = parameters_nominal_sd[i]
+
+        self._nominal_parameters_passed = True
+
+        return
+
+    def set_target_variables_nominal(
+        self,
+        target_variables_nominal_value,
+    ):
+        """Set target variables nominal value (mean)
+
+        Parameters
+        ----------
+        target_variables_nominal_value: np.array
+            An array contaning the nominal mean for target variables in
+            the order specified in target variables names at
+            initialization
+        """
+        if len(target_variables_nominal_value) != self.n_target_variables:
+            raise ValueError(
+                "Target variables array length does not match number of target variables passed at initilization."
+            )
+        for i in range(self.n_target_variables):
+            target_variable = self.target_variables_names[i]
+            self.target_variables_info[target_variable]["nominal_value"] = (
+                target_variables_nominal_value[i]
+            )
+
+        self._nominal_target_passed = True
+
+        return
+
+    def _estimate_parameter_nominal(
+        self,
+        parameters_matrix,
+    ):
+        """Estimates parameters nominal values
+
+        Parameters
+        ----------
+        parameters_matrix : np.matrix
+            Data matrix whose columns correspond to parameters values
+            ordered as passed in initialization
+
+        """
+        if parameters_matrix.shape[1] != self.n_parameters:
+            raise ValueError(
+                "Number of columns (parameters) does not match number of parameters passed at initialization."
+            )
```

Comment:
Hasn't this already been checked if there is a `parameters_matrix` ?

Reply:
You are absolutely right, this check and the one in `_estimate_target_nominal` are redundant.